### PR TITLE
avoid falling through catch and improve logs a bit

### LINKED ISSUE #267: Improve error handling
An exception should in the most cases not being catched and logged, instead rethrown as RuntimeException. There is this problem in Server, see discussion in #254 

In ReverseRequestFactory.java we do this too and catch a NPE which is ugly.

### PR DESCRIPTION
**Pull request by karussell**
Fixes #267

A few "falling through" catches were not fixed, e.g. when adding a document failed. I'm not sure if this is good, but I thought it might be that a single document fails and aborting the import process that takes hours seems ugly. But this way we silently accept potential bugs.

### CODE REVIEW

File: src/test/java/de/komoot/photon/query/ReverseRequestFactoryTest.java [hunk 1]
```diff
@@ -19,8 +19,8 @@
     private ReverseRequest reverseRequest;
 
     public void requestWithLongitudeLatitude(Request mockRequest, Double longitude, Double latitude) {
-        Mockito.when(mockRequest.queryParams("lon")).thenReturn(longitude.toString());
-        Mockito.when(mockRequest.queryParams("lat")).thenReturn(latitude.toString());
```
Comment (karussell) [inline src/test/java/de/komoot/photon/query/ReverseRequestFactoryTest.java:23]: This is a bit ugly that the tests need to know those implementation details

File: src/main/java/de/komoot/photon/elasticsearch/Server.java [hunk 2]
```diff
@@ -60,25 +57,22 @@ public MyNode(Settings preparedSettings, Collection<Class<? extends Plugin>> cla
     }
 
     public Server(CommandLineArgs args) throws Exception {
-        this(args.getCluster(), args.getDataDirectory(), args.getLanguages(), args.getTransportAddresses(), false);
+        this(args.getCluster(), args.getDataDirectory(), args.getLanguages(), args.getTransportAddresses());
     }
 
-    public Server(String clusterName, String mainDirectory, String languages, String transportAddresses,
-                  boolean isTest) throws Exception {
+    public Server(String clusterName, String mainDirectory, String languages, String transportAddresses) throws Exception {
```
Comment (lonvia) [inline src/main/java/de/komoot/photon/elasticsearch/Server.java]: Delete 'throws Exception'? Also above.

File: src/main/java/de/komoot/photon/App.java [hunk 3]
```diff
@@ -126,7 +124,7 @@ private static void startNominatimImport(CommandLineArgs args, Server esServer,
         try {
             nominatimConnector.readEntireDatabase();
         } catch (Exception e) {
-            log.info("error importing from nominatim: " + e.getMessage());
+            throw new RuntimeException("error importing from nominatim: " + e.getMessage());
```
Comment (lonvia) [inline src/main/java/de/komoot/photon/App.java]: You can remove the entire try-catch block. readEntireDatabase() only throws RuntimeExceptions in the first place.

**Comment by karussell** [comment]
Have addressed the comments

**Comment by karussell** [comment]
Would you mind to have another look @lonvia :) ?

**Comment by karussell** [comment]
> Oops, I thought you had merged it already.

:) ... just wanted to make sure that we have at least 4 eyes to approve it :)

**Pull request merged**